cmd/launcher/uninstall_darwin.go

```diff
@@ -36,13 +39,27 @@ func removeLauncher(ctx context.Context, identifier string) error {
 		fmt.Sprintf("/etc/newsyslog.d/%s.conf", identifier),
 	}
 
+	removeErr := false
+
 	// Now remove the paths used for launcher/osquery binaries and app data
 	for _, path := range pathsToRemove {
 		if err := os.RemoveAll(path); err != nil {
+			removeErr = true
 			fmt.Printf("error removing path %s: %s\n", path, err)
 		}
 	}
 
+	// If we've removed everything. unregister the pkg
+	if !removeErr {
+		pkgutiltCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+
+		cmd := exec.CommandContext(pkgutiltCtx, "/usr/sbin/pkgutil", "--forget", fmt.Sprintf("com.%s.launcher", identifier))
+		if out, err := cmd.Output(); err != nil {
+			fmt.Printf("error occurred while forgetting package: output %s: err: %s\n", out, err)
+		}
+	}
+
```

Comment:
I think we should not print a success message at the end unless it all worked?

```suggestion
        if removeErr {
                return nil
        }
        
        pkgutiltCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
	defer cancel()      
        cmd := exec.CommandContext(pkgutiltCtx, "/usr/sbin/pkgutil", "--forget", fmt.Sprintf("com.%s.launcher", identifier))
        
        if out, err := cmd.Output(); err != nil {
                fmt.Printf("error occurred while forgetting package: output %s: err: %s\n", out, err)
                return nil
	}

```